### PR TITLE
Alternative DICOM tag viewer tree

### PR DESCRIPTION
**Pull request by ferdymercury**
from https://github.com/pydicom/pydicom/pull/919

#### What contibution are you adding? How do we use it?
Adding an alternative DICOM tree viewer from the examples, which is better for checking DICOM tags than the already available one dcm_qt_tree.py viewer.

#### Any other comments?

 - [x] I have added a single script or subfolder to a topic folder
 - [x] I have added some form of documentation and my contact with the contribution
 - [x] I have added a link to the contribution in the topic folder's `README.md`

### CODE REVIEW

File: plotting-visualization/dicomtree.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+python3 dicomtree.py file.dcm
+=========================================
+Show a dicom file using hierarchical tree
+=========================================
+Show a dicom file using a hierarchical tree in a graphical window.
+"""
+
+# authors : Guillaume Lemaitre <g.lemaitre58@gmail.com>
+# license : MIT
+
```
Comment (darcymason) [inline plotting-visualization/dicomtree.py]: Please just remove the author here - we don't record them in files any more.  In any case, Guillaume was just the latest, in setting it up for online examples.  Suggest a single comment line at top with ``` # Copyright pydicom authors 2019.  See LICENSE file for details```, as is done in all pydicom files.

**Review by darcymason — COMMENTED** [review]
I haven't tried running this yet, but wanted to ask: this looks like a modification of the dicomtree.py file in the pydicom repo - perhaps it should be PR's to that repo rather than contrib-pydicom?

**Comment by darcymason** [comment]
> this looks like a modification of the dicomtree.py file in the pydicom repo - perhaps it should be PR's to that repo rather than contrib-pydicom?

Sorry - see now the reference to the pydicom issue 919 - are they the same file then?

**Comment by ferdymercury** [comment]
yes, same file

**Comment by vsoch** [comment]
hey @ferdymercury - just testing this out! I have an up to date pydicom and just installed tix-dev, I'm still hitting an issue:

```python
sudo apt install tix-dev
sudo pip install -U pydicom
python3 dicomtree.py file.dcm
=========================================
Show a dicom file using hierarchical tree
=========================================
Show a dicom file using a hierarchical tree in a graphical window.

Traceback (most recent call last):
  File "dicomtree.py", line 84, in <module>
    root = tkinter_tix.Tk()
  File "/home/vanessa/anaconda3/lib/python3.7/tkinter/tix.py", line 214, in __init__
    self.tk.eval('package require Tix')
_tkinter.TclError: can't find package Tix
```
I have Ubuntu 18.04, and I'm guessing that other users might hit this as well. Do you have any suggestions? I'd be glad to test, if I can get the script to work. If there is consistent issue, you could provide a Dockerfile alongside the image that builds the dependencies as needed.

**Comment by ferdymercury** [comment]
Thanks for your reply. I get the same error you mention in a fresh Ubuntu 18.04

The solution is to call sudo apt install tix-dev, then the error disappears. However, you said that you did it and it does not work yet.

So I guess that the difference is that you are using anaconda in a local folder /home/..., whereas I am using the default python3 installed system-wide. Can you try with the system /usr/bin/python3?

**Comment by ferdymercury** [comment]
This was my error mesage before and after:
  File "/usr/lib/python3.6/tkinter/tix.py", line 214, in __init__
    self.tk.eval('package require Tix')
_tkinter.TclError: can't find package Tix
$ sudo apt install tix-dev
Preparing to unpack .../tix_8.4.3-10_amd64.deb ...
Unpacking tix (8.4.3-10) ...
Selecting previously unselected package tix-dev.
Preparing to unpack .../tix-dev_8.4.3-10_amd64.deb ...
Unpacking tix-dev (8.4.3-10) ...
Processing triggers for libc-bin (2.27-3ubuntu1) ...
Setting up tix (8.4.3-10) ...
Processing triggers for man-db (2.8.3-2ubuntu0.1) ...
Setting up tix-dev (8.4.3-10) ...
Processing triggers for libc-bin (2.27-3ubuntu1) ...

** Now it works **

**Comment by ferdymercury** [comment]
See also https://github.com/ContinuumIO/anaconda-issues/issues/328

**Comment by vsoch** [comment]
Yep makes perfect sense, and it's funny because someone in my group was just saying that when something goes wrong on our cluster, the first thing to blame is usually conda :)

I didn't want to install to system python on my host, so I created a Singularity container recipe (we optimally use Singularity to easily use the host display without extra work):

```
Bootstrap: docker
From: ubuntu:18.04

# sudo singularity build dicomtree.sif Singularity

%files
dicomtree.py /dicomtree.py

%environment
export DEBIAN_FRONTEND=noninteractive

%post
export DEBIAN_FRONTEND=noninteractive
apt-get update && apt-get install -y tix-dev python3 python3-dev wget python3-tk
wget https://bootstrap.pypa.io/get-pip.py
python3 get-pip.py
pip3 install pydicom

%runscript
exec python3 /dicomtree.py "$@"
```
and then build:

```bash
$ sudo singularity build dicomtree.sif Singularity
```

and then run (I downloaded from the pydicom test data folder)

```bash
$ singularity run dicomtree.sif MR_small_padded.dcm 
```

Works great! 

image

So I might have missed this detail @ferdymercury  - this is a derivation of one of the pydicom scripts? How is it different?

**Comment by ferdymercury** [comment]
It allows you scrolling with the mouse wheel. Before, it was annoying be cause one had to scroll by clicking on the scroll bar on the right. That is the only difference :)
(And the window is a bit bigger.)

You also can use your file browser to open Dicom files. Just Right click on a dcm file, Open With, Other Application, Use custom command, and type python3 /path/to/dicomtree.py. Or singularity in your case.

**Comment by ferdymercury** [comment]
I just added here to this repo, because the Qt dicom viewer tree in this repo does not show DICOM tag codes.

**Comment by vsoch** [comment]
Yeah it's cool, I like it!  Given the error prone-ness of install (lots of people use conda) what do you think about creating a dicomtree folder, and adding the script, along with a README and the Singularity recipe for folks to easily use it?

**Comment by ferdymercury** [comment]
hmm, one question before, does anaconda python work with the "ttk" package? I think it is an upgrade version of "tix". So it might be easier to work with that package instead, which might remove the need of a docker around.

**Comment by vsoch** [comment]
Just a small point -  I didn't use Docker, I used Singularity. I'd be happy to test ttk with conda - how do I install it (conda install ttk didn't find a package).

**Comment by ferdymercury** [comment]
Maybe it is already there, you could try
```
from tkinter import *
from tkinter.ttk import *
Label
```

**Review by darcymason — APPROVED** [review]

**Pull request merged**